service/lib/agama/software/manager.rb

```diff
@@ -279,14 +279,23 @@ def on_selected_patterns_change(&block)
         @selected_patterns_change_callbacks << block
       end
 
-      # Determines whether a package is installed
+      # Determines whether a package is installed in the target system.
       #
       # @param name [String] Package name
       # @return [Boolean] true if it is installed; false otherwise
       def package_installed?(name)
         on_target { Yast::Package.Installed(name, target: :system) }
       end
 
+      # Determines whether a package is available.
+      #
+      # @param name [String] Package name
+      # @return [Boolean]
+      def package_available?(name)
+        # Beware: apart from true and false, Available can return nil if things go wrong.
+        on_local { !!Yast::Package.Available(name) }
```

Comment:
Be careful that `Yast::Package.Available` checks for the RPM Provides. That means the package `name` might not be available, but if there is some other package providing it then it will return `true`.

So it depends what we actually expect from the API. Maybe have two separate calls? One for package name and the other for provides?

This could cause some unexpected problems later...